.github/workflows/frontend_deploy.yaml

```diff
@@ -0,0 +1,58 @@
+name: Frontend Deploy
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
```

Comment:
Do we need to deploy on every PR?

Reply:
This creates a staging deployment for each PR so frontend changes can be viewed. It's pretty easy to just host individual staging links on vercel as there isn't really a downside.